actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -3,12 +3,27 @@ module ImplicitRender
 
     include BasicImplicitRender
 
+    # Renders the template corresponding to the controller action, if it exists.
+    # The action name, format, and variant are all taken into account.
+    # For example, the "new" action with an HTML format and variant "phone" 
+    # would try to render the <tt>new.html+phone.erb</tt> template.
+    #
+    # If no template is found <tt>ActionController::BasicImplicitRender</tt>'s implementation is called, unless
+    # a block is passed. In that case, it will override the super implementation.
+    #
+    #   default_render do
+    #     head 404 # No template was found
+    #   end
     def default_render(*args)
       if template_exists?(action_name.to_s, _prefixes, variants: request.variant)
         render(*args)
       else
-        logger.info "No template found for #{self.class.name}\##{action_name}, rendering head :no_content" if logger
-        super
+        if block_given?
+          yield(*args)
+        else
+          logger.info "No template found for #{self.class.name}\##{action_name}" if logger
```

Comment:
I think the message sounds like an error if it doesn't say that it's rendering `head :no_content`. I think it would be okay to just leave it as it was.